app/models/check_availability_task.rb

```diff
@@ -1,6 +1,9 @@
 class CheckAvailabilityTask < CloudConnectorTask
+  before_update :prevent_update, :if => proc { state_changed?(:from => 'timedout', :to => 'completed') }
```

Comment:
@hsong-rh Can this go to error state from timedout state? Since the catalog worker is setting this state it won't know that it has been marked as timedout.

Reply:
Catalog worker doesn't need to know if state is timedout or not, when it try to update `state` to `complete`, this callback will be triggered to check.